src/schema/me/credit_cards.ts

```diff
@@ -1,13 +1,18 @@
 // @ts-check
 
-import { CreditCard } from "schema/credit_card"
+import { CreditCardConnection } from "schema/credit_card"
 import { pageable } from "relay-cursor-paging"
-import { connectionWithCursorInfo } from "schema/fields/pagination"
 import { connectionFromArraySlice } from "graphql-relay"
+import { parseRelayOptions } from "lib/helpers"
+import { GraphQLInt } from "graphql"
 
 export const CreditCards = {
-  type: connectionWithCursorInfo(CreditCard.type),
-  args: pageable(),
+  type: CreditCardConnection,
+  args: pageable({
+    limit: {
```

Comment:
I'm confused, what does `limit` do? It looks like you're taking the Relay params (`first`, `after`, etc.) and parsing them to `page` and `size`. Then you're passing `page` and `size` to Gravity. I don't follow where `limit` gets used?

Reply:
That comes from `parseRelayOptions` which coerces relay params into gravity params.

I thought that `limit` was the relay-way of specifying `size`?

Reply:
`first` and `last` are size specifications, with `after` or `before` indicating the starting point and direction.

I think what you need is this field to take in a param like `first: 25` or however many we show in that UI (which it already does).

Then, you derive a `page`, `size` and `offset` from the `parseRelayOptions` helper (as you already do, just destruct `offset` as well).. You then pass `page` and `size` into Gravity (which you already do).

Then, the last step is https://github.com/artsy/metaphysics/pull/1367/files#diff-8ed13ed9a44c33b8dba29eab66141724R27 should be the total number of credit cards (derived from the `x-total-count` header, you can change the `meCreditCards` loader to return both `body` and `headers`). And then https://github.com/artsy/metaphysics/pull/1367/files#diff-8ed13ed9a44c33b8dba29eab66141724R28 should be the `offset` you destructed in `parseRelayOptions`.

You can drop `limit` too.